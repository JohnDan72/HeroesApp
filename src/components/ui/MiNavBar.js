import React, { useContext } from 'react';
// import PropTypes from 'prop-types';
import { Nav, Navbar } from 'rsuite';
import { CustomNavLink } from './CustomNavLink';
import { GeneralContext } from '../../GeneralContext';

// react icons
import { BsFillPersonFill } from "react-icons/bs";
import { SiLooker } from "react-icons/si";
import { GiBatMask, GiMaskedSpider } from 'react-icons/gi';
import { MdColorLens, MdDarkMode, MdLightMode } from 'react-icons/md';



const propTypes = {};
const defaultProps = {};

const MiNavBar = () => {
    const { setTheme, active, setActive } = useContext(GeneralContext);

    return (

        <Navbar >
            <div className="container">
                <Navbar.Brand as={CustomNavLink} href="/">
                    Heroes App
                </Navbar.Brand>
                <Nav activeKey={active} onSelect={setActive}>
                    <Nav.Item as={CustomNavLink} href="/dc" eventKey="dc" icon={<GiBatMask />}>DC</Nav.Item>
                    <Nav.Item as={CustomNavLink} href="/marvel" eventKey="marvel" icon={<GiMaskedSpider />}>Marvel</Nav.Item>
                    <Nav.Item as={CustomNavLink} href="/search" eventKey="search" icon={<SiLooker />}>Search</Nav.Item>

                </Nav>
                <Nav pullRight activeKey={active} onSelect={setActive}>
                    <Nav.Dropdown title="Theme" icon={<MdColorLens />}>
                        <Nav.Dropdown.Item as={CustomNavLink} icon={<MdLightMode />} eventKey='light' onClick={() => setTheme('light')}> Mode light</Nav.Dropdown.Item>
                        <Nav.Dropdown.Item as={CustomNavLink} icon={<MdDarkMode />} eventKey='dark' onClick={() => setTheme('dark')}> Mode dark</Nav.Dropdown.Item>
                    </Nav.Dropdown>
                    <Nav.Item as={CustomNavLink} href="/login" icon={<BsFillPersonFill />}>Logout</Nav.Item>
                </Nav>

            </div>
        </Navbar>
    );
}

MiNavBar.propTypes = propTypes;
MiNavBar.defaultProps = defaultProps;

export default MiNavBar;