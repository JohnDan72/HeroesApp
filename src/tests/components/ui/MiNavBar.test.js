import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import MiNavBar from "../../../components/ui/MiNavBar";
import { GeneralContext } from "../../../GeneralContext";
import { types } from "../../../types/types";

// mock antes de la variable es importante para que salga bien la tarea
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}))


describe('Pruebas en MiNavBar', () => {
    const contextValue = {
        user: {
            name: 'Juancho'
        },
        active: 'search',
        setActive: jest.fn(),
        theme: 'dark',
        setTheme: jest.fn(),
        dispatch: jest.fn()
    }

    test('debe mostrar el component correctamente', () => {
        const wrapper = mount(
            <GeneralContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/marvel']}>
                    <MiNavBar />
                </MemoryRouter>
            </GeneralContext.Provider>
        );
        const nav_account = wrapper.find('#id_account_nav');
        expect(wrapper).toMatchSnapshot();
        expect(nav_account.at(0).prop('title').trim()).toBe(contextValue.user.name);
    });

    test('debe llamar logout, llamar navigate y dispatch con argumentos', () => {
        const wrapper = mount(
            <GeneralContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/marvel']}>
                    <MiNavBar />
                </MemoryRouter>
            </GeneralContext.Provider>
        );
        const logout = wrapper.find('#id_logout');

        // console.log(logout.length);
        // console.log(logout.at(logout.length - 1).props());

        logout.at(logout.length - 1).simulate('click');

        expect(mockNavigate).toHaveBeenCalledTimes(1);
        expect(mockNavigate).toHaveBeenCalledWith('/login');
        expect(contextValue.dispatch).toHaveBeenCalledTimes(1);
        expect(contextValue.dispatch).toHaveBeenCalledWith({type: types.logout});
        
    });


})
