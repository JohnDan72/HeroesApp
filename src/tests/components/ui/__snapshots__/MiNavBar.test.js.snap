// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MiNavBar debe mostrar el component correctamente 1`] = `
<nav
  className="rs-navbar rs-navbar-inverse"
>
  <div
    className="container"
  >
    <a
      className="logoDiv rs-navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height="50"
        src="logoCircle.png"
      />
    </a>
    <div
      className="rs-navbar-nav rs-nav rs-nav-default rs-nav-horizontal"
    >
      <a
        className="rs-navbar-item"
        href="/dc"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 16c48 32 64 80 64 112 0 128 0 256-32 368l-16-112c-48 32-112 64-144 64s-96-32-144-64L96 496C64 384 64 256 64 128c0-32 16-80 64-112-16 48-16 96 0 128 64-32 192-32 256 0 16-32 16-80 0-128zm0 256c-33.7 46.8-52 57.2-112 80 33.5 32 95.5 21.5 128-16 14.1-16.3 6.8-44.6-16-64zm-256 0c-22.8 19.4-30.1 47.7-16 64 32.5 37.5 94.5 48 128 16-60-22.8-78.3-33.2-112-80z"
            key="0"
          />
        </svg>
        DC
        <span
          className="rs-ripple-pond"
        >
          <span
            className="rs-ripple"
          />
        </span>
      </a>
      <a
        className="rs-navbar-item"
        href="/marvel"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M255.9 18.98c-23.8 0-48.4 22.04-66.8 53.04-18.3 30.98-29.6 69.88-29.6 96.08 0 32 16 60.4 40.7 77.6 14.4-17.3 34-28.4 55.7-28.4 21.7 0 41.2 11.1 55.6 28.4 24.7-17.2 40.8-45.6 40.8-77.6 0-26.2-11.4-65.1-29.7-96.08-18.3-31-42.9-53.04-66.7-53.04zm237.6 14.14L419.8 207.1 324 264.4c2.8 5.4 5.3 11.2 7.4 17.3L434.6 220l58.9-139zm-475.24 0V81l58.92 139 103.22 61.5c2.1-6.1 4.6-11.9 7.5-17.3L92 207.1c-24.59-58-49.15-116.01-73.74-173.97zM255.9 236c-16.9 0-32.6 9.3-44.6 25.9-12 16.5-19.8 40-19.8 66.2 0 15.4 2.7 29.8 7.3 42.4 3.1-5.4 9-9.1 15.8-9.1 10.1 0 18.2 8.1 18.2 18.1 0 10-8.1 18.1-18.2 18.1-.3 0-.6-.1-.9-.1 11.7 14.6 26.4 22.7 42.2 22.7 15.7 0 30.5-8.1 42.1-22.7-.3 0-.6.1-.9.1-10 0-18.2-8.1-18.2-18.1 0-10 8.2-18.1 18.2-18.1 6.8 0 12.7 3.7 15.8 9.1 4.7-12.6 7.4-27 7.4-42.4 0-26.2-7.8-49.7-19.8-66.2-12.1-16.6-27.7-25.9-44.6-25.9zm-237.65 1.6v23.9l64.5 51.2 90.05 12.5c.1-6.4.7-12.6 1.5-18.7l-83.93-11.6-72.11-57.3zm475.25 0l-72.1 57.3-84 11.7c.9 6.1 1.4 12.3 1.5 18.7l90.1-12.6 64.5-51.2zm-263.6 81.1c10 0 18.2 8.1 18.2 18.1 0 10-8.2 18.1-18.2 18.1-10.1 0-18.2-8.1-18.2-18.1 0-10 8.1-18.1 18.2-18.1zm52 0c10 0 18.2 8.1 18.2 18.1 0 10-8.2 18.1-18.2 18.1-10.1 0-18.2-8.1-18.2-18.1 0-10 8.1-18.1 18.2-18.1zm55.3 32.3c-1 6.3-2.4 12.3-4.1 18.2l88 19.2 72.4 58.3v-24l-64.2-51.6zm-162.6.1l-92.18 20-64.16 51.6v24l72.44-58.3 88-19.1c-1.8-5.9-3.2-12-4.1-18.2zm149.9 39.7c-3.2 5.9-6.4 11.2-9.9 15.9l67.8 36 27.5 51h21.3l-35-64.9zm-137.2.1l-71.7 37.9-35.01 64.9h21.21l27.5-51 68-35.9c-3.6-5.1-7.4-10.8-10-15.9zm20.3 27.5c-5.4 8.7-8.6 18.9-8.6 29.8 0 15 5.9 28.6 15.4 38.7v-2.5c0-20.9 8.1-39.2 20.2-49.2-9.9-3.5-19.1-9.3-27-16.8zm96.6 0c-8 7.5-17.1 13.3-27 16.8 12.1 10 20.1 28.3 20.1 49.2v2.5c9.6-10.1 15.4-23.7 15.4-38.7 0-10.9-3.1-21.1-8.5-29.8z"
            key="0"
          />
        </svg>
        Marvel
        <span
          className="rs-ripple-pond"
        >
          <span
            className="rs-ripple"
          />
        </span>
      </a>
      <a
        aria-selected={true}
        className="rs-navbar-item rs-navbar-item-active"
        href="/search"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
            key="0"
          />
        </svg>
        Search
        <span
          className="rs-ripple-pond"
        >
          <span
            className="rs-ripple"
          />
        </span>
      </a>
    </div>
    <div
      className="rs-navbar-nav rs-navbar-right rs-nav rs-nav-default rs-nav-horizontal"
    >
      <div
        className="rs-dropdown rs-dropdown-placement-bottom-start"
      >
        <button
          aria-expanded={false}
          className="rs-dropdown-toggle rs-btn rs-btn-default"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          type="button"
        >
          <svg
            className="rs-dropdown-toggle-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              key="0"
            />
            <path
              d="M12 3a9 9 0 000 18c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
              key="1"
            />
          </svg>
          Theme
          <svg
            aria-hidden={true}
            aria-label="angle down"
            className="rs-dropdown-toggle-caret rs-icon"
            data-category="legacy"
            fill="currentColor"
            focusable={false}
            height="1em"
            viewBox="0 0 21 32"
            width="1em"
          >
            <path
              d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
            />
          </svg>
          <span
            className="rs-ripple-pond"
          >
            <span
              className="rs-ripple"
            />
          </span>
        </button>
        <ul
          className="rs-dropdown-menu"
          hidden={true}
        >
          <li>
            <a
              className="rs-dropdown-item rs-dropdown-item-with-icon"
              data-event-key="light"
              href="/marvel"
              onClick={[Function]}
              role="button"
            >
              <svg
                className="rs-dropdown-item-menu-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  key="0"
                />
                <path
                  d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 000-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
                  key="1"
                />
              </svg>
               Mode light
            </a>
          </li>
          <li>
            <a
              className="rs-dropdown-item rs-dropdown-item-with-icon"
              data-event-key="dark"
              href="/marvel"
              onClick={[Function]}
              role="button"
            >
              <svg
                className="rs-dropdown-item-menu-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  key="0"
                />
                <path
                  d="M12 3a9 9 0 109 9c0-.46-.04-.92-.1-1.36a5.389 5.389 0 01-4.4 2.26 5.403 5.403 0 01-3.14-9.8c-.44-.06-.9-.1-1.36-.1z"
                  key="1"
                />
              </svg>
               Mode dark
            </a>
          </li>
        </ul>
      </div>
      <div
        className="rs-dropdown rs-dropdown-placement-bottom-start"
      >
        <button
          aria-expanded={false}
          className="rs-dropdown-toggle rs-btn rs-btn-default"
          id="id_account_nav"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          type="button"
        >
          <svg
            className="rs-dropdown-toggle-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              key="0"
            />
          </svg>
          Juancho
          <svg
            aria-hidden={true}
            aria-label="angle down"
            className="rs-dropdown-toggle-caret rs-icon"
            data-category="legacy"
            fill="currentColor"
            focusable={false}
            height="1em"
            viewBox="0 0 21 32"
            width="1em"
          >
            <path
              d="M19.196 13.143a.612.612 0 01-.179.411l-8.321 8.321c-.107.107-.268.179-.411.179s-.304-.071-.411-.179l-8.321-8.321c-.107-.107-.179-.268-.179-.411s.071-.304.179-.411l.893-.893a.582.582 0 01.411-.179c.143 0 .304.071.411.179l7.018 7.018 7.018-7.018c.107-.107.268-.179.411-.179s.304.071.411.179l.893.893a.617.617 0 01.179.411z"
            />
          </svg>
          <span
            className="rs-ripple-pond"
          >
            <span
              className="rs-ripple"
            />
          </span>
        </button>
        <ul
          className="rs-dropdown-menu"
          hidden={true}
        >
          <li>
            <a
              className="rs-dropdown-item rs-dropdown-item-with-icon"
              href="/marvel"
              id="id_logout"
              onClick={[Function]}
              role="button"
            >
              <svg
                className="rs-dropdown-item-menu-icon"
                fill="none"
                height="1em"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                  key="0"
                />
                <polyline
                  key="1"
                  points="16 17 21 12 16 7"
                />
                <line
                  key="2"
                  x1="21"
                  x2="9"
                  y1="12"
                  y2="12"
                />
              </svg>
              Logout
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</nav>
`;
